cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControlUtils.java

```diff
@@ -430,13 +430,13 @@ public static MetadataResponse prepareMetadataResponse(List<Node> brokers,
     for (MetadataResponse.PartitionMetadata partitionMetadata : topicMetadata.partitionMetadata()) {
       metadataResponseTopic.partitions().add(
           new MetadataResponseData.MetadataResponsePartition()
-              .setErrorCode(partitionMetadata.error().code())
+              .setErrorCode(partitionMetadata.error.code())
               .setPartitionIndex(partitionMetadata.partition())
-              .setLeaderId(partitionMetadata.leader() == null ? -1 : partitionMetadata.leader().id())
-              .setLeaderEpoch(partitionMetadata.leaderEpoch().orElse(RecordBatch.NO_PARTITION_LEADER_EPOCH))
-              .setReplicaNodes(partitionMetadata.replicas().stream().map(Node::id).collect(Collectors.toList()))
-              .setIsrNodes(partitionMetadata.isr().stream().map(Node::id).collect(Collectors.toList()))
-              .setOfflineReplicas(partitionMetadata.offlineReplicas().stream().map(Node::id).collect(Collectors.toList())));
+              .setLeaderId(partitionMetadata.leaderId.orElse(-1))
```

Comment:
-1 could be MetadataResponse.NO_LEADER_ID

Reply:
thanks - updated